source/cellular_pktio.c

```diff
@@ -273,24 +302,65 @@ static CellularATCommandResponse_t * _Cellular_AtResponseNew( void )
  */
 static void _Cellular_AtResponseFree( CellularATCommandResponse_t * pResp )
 {
-    CellularATCommandLine_t * pCurrLine = NULL;
-    CellularATCommandLine_t * pToFree = NULL;
+    #if ( CELLULAR_CONFIG_STATIC_AT_RESPONSE == 1 )
+        ( void ) pResp;
+        cellularStaticATCommandIndex = 0;
+    #else
+        CellularATCommandLine_t * pCurrLine = NULL;
+        CellularATCommandLine_t * pToFree = NULL;
+
+        if( pResp != NULL )
+        {
+            pCurrLine = pResp->pItm;
 
-    if( pResp != NULL )
-    {
-        pCurrLine = pResp->pItm;
+            while( pCurrLine != NULL )
+            {
+                pToFree = pCurrLine;
+                pCurrLine = pCurrLine->pNext;
 
-        while( pCurrLine != NULL )
-        {
-            pToFree = pCurrLine;
-            pCurrLine = pCurrLine->pNext;
+                /* Ruese the pktiobuffer. No need to free pToFree->pLine here. */
+                Platform_Free( pToFree );
+            }
+
+            Platform_Free( pResp );
+        }
+    #endif /* if ( CELLULAR_CONFIG_STATIC_AT_RESPONSE == 1 ) */
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * Allocate CellularATCommandLine_t structure. The allocated memory is freed in
+ *  _Cellular_AtResponseFree.
+ */
+static CellularATCommandLine_t * _Cellular_AtCommandLineNew( void )
+{
+    CellularATCommandLine_t * pNew = NULL;
 
-            /* Ruese the pktiobuffer. No need to free pToFree->pLine here. */
-            Platform_Free( pToFree );
+    #if ( CELLULAR_CONFIG_STATIC_AT_RESPONSE == 1 )
+        if( cellularStaticATCommandIndex < CELLULAR_NUM_AT_COMMAND_LINE )
+        {
+            pNew = &cellularStaticATCommandLineTable[ cellularStaticATCommandIndex ];
+            cellularStaticATCommandIndex++;
```

Comment:
Is it possible that multi-threads request to send AT command at the same time?
Do we need to handle this race condition?